crates/trigger-redis/src/lib.rs

```diff
@@ -1,203 +1,226 @@
-pub(crate) type Store = spin_core::Store<RuntimeData>;
-
-type ChannelComponents = HashMap<String, Vec<String>>;
-/// The Spin Redis trigger.
-#[derive(Clone)]
-pub struct RedisTrigger {
-    engine: Arc<TriggerAppEngine<Self>>,
-    // Mapping of server url with subscription channel and associated component IDs
-    server_channels: HashMap<String, ChannelComponents>,
+/// Redis trigger metadata.
+#[derive(Clone, Debug, Default, Deserialize)]
+#[serde(deny_unknown_fields)]
+struct TriggerMetadata {
+    address: String,
 }
 
 /// Redis trigger configuration.
-#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+#[derive(Clone, Debug, Default, Deserialize)]
 #[serde(deny_unknown_fields)]
-pub struct RedisTriggerConfig {
+struct TriggerConfig {
     /// Component ID to invoke
-    pub component: String,
+    component: String,
     /// Channel to subscribe to
-    pub channel: String,
-    /// optional overide address for trigger
-    pub address: Option<String>,
-    /// Trigger executor (currently unused)
-    #[serde(default, skip_serializing)]
-    pub executor: IgnoredAny,
+    channel: String,
+    /// Optionally override address for trigger
+    address: Option<String>,
 }
 
-#[derive(Clone, Debug, Default, Deserialize, Serialize)]
-#[serde(deny_unknown_fields)]
-struct TriggerMetadata {
-    address: String,
-}
+impl Trigger for RedisTrigger {
+    const TYPE: &'static str = "redis";
 
-#[async_trait]
-impl TriggerExecutor for RedisTrigger {
-    const TRIGGER_TYPE: &'static str = "redis";
-    type RuntimeData = RuntimeData;
-    type TriggerConfig = RedisTriggerConfig;
-    type RunConfig = NoArgs;
-    type InstancePre = InstancePre<RuntimeData>;
-
-    async fn new(engine: TriggerAppEngine<Self>) -> Result<Self> {
-        let default_address: String = engine
-            .trigger_metadata::<TriggerMetadata>()?
-            .unwrap_or_default()
-            .address;
-        let default_address_expr = spin_expressions::Template::new(default_address)?;
-        let default_address = engine.resolve_template(&default_address_expr)?;
-
-        let mut server_channels: HashMap<String, ChannelComponents> = HashMap::new();
-
-        for (_, config) in engine.trigger_configs() {
-            let address = config.address.clone().unwrap_or(default_address.clone());
-            let address_expr = spin_expressions::Template::new(address)?;
-            let address = engine.resolve_template(&address_expr)?;
-            let server = server_channels.entry(address).or_default();
-            let channel_expr = spin_expressions::Template::new(config.channel.as_str())?;
-            let channel = engine.resolve_template(&channel_expr)?;
-            server
-                .entry(channel)
-                .or_default()
-                .push(config.component.clone());
-        }
-        Ok(Self {
-            engine: Arc::new(engine),
-            server_channels,
-        })
+    type CliArgs = NoCliArgs;
+
+    type InstanceState = ();
```

Comment:
Random thought: I like how `type CliArgs = NoCliArgs;` reads much better than `type InstanceState = ();`. Perhaps we should consider a `NoInstanceState` type which could be used for empty instance state instead of `()`.

Reply:
fwiw `NoCliArgs` is only necessary because of the `type CliArgs: clap::Args` bound.